skfp/model_selection/scaffold_split.py

```diff
@@ -0,0 +1,365 @@
+import warnings
+from collections import defaultdict
+from collections.abc import Sequence
+from itertools import chain
+from numbers import Integral
+from typing import Any, Optional, Union
+
+import numpy as np
+from rdkit.Chem import Mol
+from rdkit.Chem.Scaffolds import MurckoScaffold
+from sklearn.utils._param_validation import Interval, RealNotInt, validate_params
+
+from skfp.model_selection.utils import (
+    ensure_nonempty_list,
+    fill_missing_sizes,
+    get_data_from_indices,
+    split_additional_data,
+    validate_train_valid_test_split_sizes,
+)
+from skfp.utils.validators import ensure_mols
+
+
+@validate_params(
+    {
+        "data": ["sequence"],
+        "additional_data": ["list"],
+        "train_size": [
+            Interval(RealNotInt, 0, 1, closed="neither"),
+            Interval(Integral, 1, float("inf"), closed="left"),
```

Comment:
Does `np.inf` work? It would be a bit better than converting string

Reply:
It does, for some reason I haven't considered

Reply:
My mistake; the original scikit-learn implementation uses 'None' here.